langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java

```diff
@@ -109,9 +135,14 @@ private static void checkIndexWithinRange(BString str, long startIndex) {
         }
 
         int strLength = str.length();
-        if (strLength <= startIndex) {
+        if (strLength != 0 && strLength <= startIndex) {
             throw BLangExceptionHelper.getRuntimeException(BallerinaErrorReasons.INDEX_OUT_OF_RANGE_ERROR,
                     RuntimeErrors.INVALID_REGEXP_FIND_INDEX, startIndex, strLength);
         }
     }
+
+    private static boolean isEmptyRegexp(BRegexpValue regExp) {
+
```

Comment:
```suggestion
```

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/39444/commits/d2e825dc93401a2bcd7bac4ee797b9304347c3ee.